movement/analysis/kinematics.py

```diff
@@ -165,3 +169,472 @@ def compute_time_derivative(data: xr.DataArray, order: int) -> xr.DataArray:
+        or ``keypoints``. The key is the keypoint or individual and the
+        value is a list of keypoints or individuals or a string
+        representing a keypoint or individual to compute the distance with.
+        If not provided, defaults to ``None`` and all possible combinations
+        of pairs are computed.
+    metric : str, optional
+        The distance metric to use. Must be one of the options supported
+        by :func:`scipy.spatial.distance.cdist`, i.e.
+        ``'braycurtis'``, ``'canberra'``, ``'chebyshev'``, ``'cityblock'``,
+        ``'correlation'``, ``'cosine'``, ``'dice'``, ``'euclidean'``,
+        ``'hamming'``, ``'jaccard'``, ``'jensenshannon'``, ``'kulczynski1'``,
+        ``'mahalanobis'``, ``'matching'``, ``'minkowski'``,
+        ``'rogerstanimoto'``, ``'russellrao'``, ``'seuclidean'``,
+        ``'sokalmichener'``, ``'sokalsneath'``, ``'sqeuclidean'``, ``'yule'``.
+        Defaults to ``'euclidean'``.
+    **kwargs : dict
+        Additional keyword arguments to pass to
+        :func:`scipy.spatial.distance.cdist`.
+
+    Returns
+    -------
+    xarray.DataArray | dict[str, xarray.DataArray]
+        An :class:`xarray.DataArray` containing the computed distances between
+        a single pair of ``individuals`` or ``keypoints``. If multiple
+        ``pairs`` are provided, this will be a dictionary containing the
+        computed distances for each pair, with the key being the pair of
+        keypoints or individuals and the value being the
+        :class:`xarray.DataArray` containing the computed distances.
+
+    See Also
+    --------
+    cdist : ``movement``'s wrapper around :func:`scipy.spatial.distance.cdist`.
+    :func:`scipy.spatial.distance.cdist` : The underlying function used.
+
+    """
+    if dim not in ["individuals", "keypoints"]:
+        raise log_error(
+            ValueError,
+            "'dim' must be either 'individuals' or 'keypoints', "
+            f"but got {dim}.",
+        )
+    pairwise_distances = {}
+
+    # Compute all possible pair combinations if not provided
+    if pairs is None:
+        paired_elements = list(
+            itertools.combinations(getattr(data, dim).values, 2)
+        )
+    else:
+        paired_elements = [
+            (elem1, elem2)
+            for elem1, elem2_list in pairs.items()
+            for elem2 in (
+                [elem2_list] if isinstance(elem2_list, str) else elem2_list
+            )
+        ]
+    for elem1, elem2 in paired_elements:
+        input1 = data.sel({dim: elem1})
+        input2 = data.sel({dim: elem2})
+        pairwise_distances[f"dist_{elem1}_{elem2}"] = cdist(
+            input1, input2, dim=dim, metric=metric, **kwargs
+        )
+    # Return DataArray if result only has one key
+    if len(pairwise_distances) == 1:
+        return next(iter(pairwise_distances.values()))
+    return pairwise_distances
+
+
+def _validate_core_dimension(
```

Comment:
I find the `core_dimension` term quite confusing. I constantly kept mixing `core_dim` and `dim` (as in the dim across which the distance is measured) in my mind - which baffled me on my initial read of this function.

I know you'be taken this term from `xr.apply_ufunc` but I am wondering if there is some other way to call it.

Reply:
Changed this to `labels_dimension` / `labels_dim`  to mean the dimension from which ``dim`` labels are obtained.

Reply:
That's better, thanks.